### PR TITLE
added env and DB stuff

### PR DESCRIPTION
**Pull request by aalexcomas11**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import Koa from 'koa'
 import sampleAPI from './API/sampleAPI'
 const app = new Koa()
```
Comment (Jehfkemsy) [inline src/app.ts]: Highly recommend the followings:
- might wanna consider a logger for debugging production code
- `koa-bodyparser` for automatically parsing of `ctx.body`
- `koa-cors` for sharing resources across different origin (which I'm pretty sure you're going to be doing) 
- `koa-helmet` for security headers

File: src/pubsub/pubSub.ts [hunk 2]
```diff
@@ -1,51 +1,47 @@
...
 export function createMessageBus(): MessageBus {
   const cachedTopics: Map<string, Topic> = new Map()
   const cachedShellSubscription: Map<string, Subscription> = new Map()
 
-  const pubSub = new PubSub({ projectId: PROJECT_ID })
-  const nameParser = (name: string, type: 'subscriptions' | 'topics'): string =>
-    name.replace(`projects/${PROJECT_ID}/${type}/`, '')
+  const pubSub = new PubSub({
+    projectId: process.env.PROJECT_ID,
+    keyFilename: path.join(__dirname, '/../../pubsub.json'),
```
[the remark this answers is not in the document]
Comment (aalexcomas11) [reply src/pubsub/pubSub.ts]: I tried it and I was getting a typescipt specific issue and did not have bandwidth to solve it.

File: src/pubsub/pubSub.ts [hunk 3]
```diff
@@ -1,51 +1,47 @@
...
+        const [theTopic] = await pubSub.createTopic(topicName)
+        cachedTopics.set(nameParser(theTopic.name), theTopic)
       }
-      const [theTopic] = await pubSub.createTopic(topicName)
-      cachedTopics.set(nameParser(theTopic.name, 'topics'), theTopic)
-      return theTopic
+      return cachedTopics.get(topicName)!
     } catch (err) {
       const error = `Failed to retrieve topic ${topicName} - ${err.message}`
       console.log(error)
```
[the remark this answers is not in the document]
Comment (aalexcomas11) [reply src/pubsub/pubSub.ts]: I did have it in there before, must of removed it when I was refactoring

File: src/app.ts (hunk 1 above)
Comment (aalexcomas11) [reply src/app.ts]: Noted

File: src/app.ts [hunk 4]
```diff
@@ -2,8 +2,9 @@ import Koa from 'koa'
...
 import mongoose from 'mongoose'
+require('dotenv').config()
```
[the remark this answers is not in the document]
Comment (aalexcomas11) [reply src/app.ts]: Tried that, did not work for me lol. I'll move it to the top

File: src/app.ts (hunk 1 above)
Comment (aalexcomas11) [reply src/app.ts]: I'll install them, but will not have samples on the template for them because of time constraints.

**Pull request merged**